src/Lachain.Core/RPC/HTTP/Web3/AccountServiceWeb3.cs

```diff
@@ -39,8 +39,42 @@ public class AccountServiceWeb3 : JsonRpcService
         public string GetBalance(string address, string tag)
         {
             var addressUint160 = address.HexToUInt160();
-            var availableBalance = GetSnapshotByTag(tag)!.Balances.GetBalance(addressUint160);
-            return Web3DataFormatUtils.Web3Number(availableBalance.ToWei().ToUInt256());
+
+            if (tag == "pending")
+            {
+                //Extract all the transactions from the pool
+                //Extract the transactions where 'to' is this address(let's say type-1)
+                //Extract the transactions where 'from' is this address(let's say type-2)
+                //Then you have to simulate execution of all these transactions in the(ReceiptCompare Order).
+                //Then you can find the pending balance of the address.
```

Comment:
The comments are a bit misleading. We should comment the version we are considering.